### PR TITLE
Version 1.0

### LINKED ISSUE #1: Tasks To Be Done for v1
These are the works that should be done before our first release:
- [x] Error handling in each function
- [x] Add `final_dist` method for calculating final distirbution of the random walk.
- [x] Tests for graphical part and macOS
- [x] Making a `.ipynb` document and its tests
- [ ] Version releasing tasks
- [x] Add codefactor
- [x] Add related badeges to `README.md`

### LINKED ISSUE #3: Add Pay-off Cost Random Walks
#### Description
There is a major category in random walk which you will be payed off according to given `payoff` list by stopping your walk in each state and/or will pay a cost according to given `cost` variable. (You may take a factor called `discount` into consider which will be manipulate our payoff by multiplying into it. [0<`discount`<=1])

```
>>> rw = RandomWalk([0, 1, 2],
...                [[0.5, 0.5, 0], [0.25, 0.5, 0.25], [0, 0, 0]],
...                payoff=[1, 4, 0],
...                cost=0.5,
...                discount=0.9)
>>> rw.best_policy()
```

### PR DESCRIPTION
**Pull request by sadrasabouri**
#### Reference Issues/PRs
This PR will close #1 and close #3.
This version contains:
```
### Added
- codefactor tests
- pypi installing guide
- `Document.ipynb`
- `Examples.ipynb`
- `notebook_check.py`
- `precision` added to `is_prob_dist`
- `payoff` attribute
- `cost` attribute
- `discount` attribute
- `best_policy` method
### Changes
- `make_prob_dist` bug fixed
- `final_dist` bug fixed
```

### CODE REVIEW

**Comment by sepandhaghighi** [comment]
@sadrasabouri Nice 💯 
I will review this PR ASAP

File: README.md [hunk 1]
```diff
@@ -238,6 +251,19 @@ array([[0.  , 0.75, 0.  , 0.  ],
...
+```pycon
+>>> states = [0, 1, 2]
+>>> trans = np.array([[1, 0, 0], [1/2, 0, 1/2], [0, 1, 0]])
+>>> rw = RandomWalk(states, trans, payoff=[0, 1, 4], cost=[1, 0, 2], discount=0.5)
+>>> rw.best_policy(stop_states=[0])
+{'continue': [], 'stop': [0, 1, 2]}
+```
+
+
 ## References
```
Comment (sepandhaghighi) [inline README.md:268]: Correct format:
<blockquote>1- Lawler, Gregory F. Introduction to stochastic processes. Chapman and Hall/CRC, 2018.</blockquote>
<blockquote>2- <a href="https://markusfeng.com/projects/graph/">Markusfeng</a></blockquote>

**Review by sepandhaghighi — CHANGES_REQUESTED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -20,5 +34,5 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 - `get_typeof_classes` method
 - `README.md`
 
-[Unreleased]: https://github.com/sadrasabouri/pyrandwalk/compare/v0.09...dev
-
+[Unreleased]: https://github.com/sadrasabouri/pyrandwalk/compare/v1.0...dev
```
Comment (sepandhaghighi) [inline CHANGELOG.md:37]: What? PyCM?!?

Correct:
```
[Unreleased]: https://github.com/sadrasabouri/pyrandwalk/compare/v1.0...dev
[1.0]: https://github.com/sadrasabouri/pyrandwalk/compare/v0.09...v1.0
[0.09]: https://github.com/sadrasabouri/pyrandwalk/compare/9113e4c...v0.09
```

**Review by sepandhaghighi — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 2 above)
Comment (sadrasabouri) [reply CHANGELOG.md:37]: Oh :+1: 
Copy paste problems :smile:.

**Review by sepandhaghighi — CHANGES_REQUESTED** [review]
Please take a look at `CONTRIBUTING.md`.
```
To add a new class statistic visit here
To add a new overall statistic visit here
To add a new interpretation visit here
```

**Comment by sadrasabouri** [comment]
> Please take a look at `CONTRIBUTING.md`.
> 
> ```
> To add a new class statistic visit here
> To add a new overall statistic visit here
> To add a new interpretation visit here
> ```

Aha. Same copy paste problem.

File: CHANGELOG.md [hunk 3]
```diff
@@ -5,7 +5,21 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
...
 ## [Unreleased]
-
+## [1.0] - 2021-06-09
+### Added
+- codefactor tests
+- pypi installation guide
+- `Document.ipynb`
+- `Examples.ipynb`
+- `notebook_check.py`
+- `precision` added to `is_prob_dist`
```
Comment (sepandhaghighi) [inline CHANGELOG.md]: `precision` (parameter?) added to `is_prob_dist` (function?)

File: Document/Document.ipynb [hunk 4]
```diff
@@ -0,0 +1,808 @@
...
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Requirement already satisfied: decorator>=4.3.0 in /usr/local/lib/python3.8/dist-packages (from networkx>=2->pyrandwalk) (4.4.2)\r\n"
+     ]
+    }
+   ],
+   "source": [
+    "!pip install pyrandwalk\n",
```
Comment (sepandhaghighi) [inline Document/Document.ipynb]: Run `!pip -q -q install pyrandwalk` to bypass the installation messages

File: Document/Examples.ipynb [hunk 5]
```diff
@@ -0,0 +1,197 @@
...
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Requirement already satisfied: decorator>=4.3.0 in /usr/local/lib/python3.8/dist-packages (from networkx>=2->pyrandwalk) (4.4.2)\r\n"
+     ]
+    }
+   ],
+   "source": [
+    "!pip install pyrandwalk\n",
```
Comment (sepandhaghighi) [inline Document/Examples.ipynb]: Run `!pip -q -q install pyrandwalk` to bypass the installation messages

**Review by sepandhaghighi — CHANGES_REQUESTED** [review]

**Review by sepandhaghighi — APPROVED** [review]

**Comment by sepandhaghighi** [comment]
@sadrasabouri LGTM 🥇

**Pull request merged**